components/sup/src/manager/mod.rs

```diff
@@ -613,6 +629,19 @@ impl Suitability for SuitabilityLookup {
     }
 }
 
+fn obtain_process_lock(fs_cfg: &FsCfg) -> Result<()> {
+    match OpenOptions::new().write(true).create_new(true).open(&fs_cfg.proc_lock_file) {
+        Ok(_) => Ok(()),
+        Err(_) => Err(sup_error!(Error::NoProcessLock)),
```

Comment:
Here we should be checking to see if the PID mentioned in the lock file is present, and if it is not, we should release the lock and try again.